.docker/Dockerfile

```diff
@@ -38,29 +44,29 @@ RUN apt update && \
     rosdep init && \
     rosdep update
 
-WORKDIR $WORKSPACE
+# Setup the workspace
+WORKDIR ${WORKSPACE}
 
-RUN git clone https://github.com/moveit/moveit2.git -b main && \
-    for repo in moveit2/moveit2.repos $(f="moveit2/moveit2_$ROS_DISTRO.repos"; \
-    test -r $f && echo $f); do vcs import < "$repo"; done  && \
-    apt-get update && \
-    rosdep install -r --from-paths . --ignore-src --rosdistro $ROS_DISTRO -y
+RUN mkdir -p src && \
+    cd src && \
+    git clone https://github.com/moveit/moveit2.git -b main && \
+    for repo in moveit2/moveit2.repos \
+        $(f="moveit2/moveit2_${ROS_DISTRO}.repos"; test -r $f && echo $f); \
+        do vcs import < "$repo"; done && \
+    apt update && \
+    . /opt/ros/${ROS_DISTRO}/setup.sh && \
+    rosdep install -r --from-paths . --ignore-src --rosdistro ${ROS_DISTRO} -y
 
 RUN . /opt/ros/${ROS_DISTRO}/setup.sh && \
-    apt remove ros-$ROS_DISTRO-moveit*
-
-WORKDIR $WORKSPACE/..
-
-RUN . /opt/ros/$ROS_DISTRO/setup.sh && \
-    apt-get update && \
-    apt-get install -y ros-${ROS_DISTRO}-rmw-cyclonedds-cpp && \
-    rosdep update && \
-    DEBIAN_FRONTEND=noninteractive \
-    rosdep install --from-paths src --ignore-src -r -y --rosdistro \
-    ${ROS_DISTRO} && \
     colcon build --mixin release --parallel-workers 1
 
-# create entrypoint
+# Install any other dependencies of this package
+COPY . ${WORKSPACE}/src/moveit_drake
+RUN apt update && \
+    . /opt/ros/${ROS_DISTRO}/setup.sh && \
+    rosdep install -r --from-paths src --ignore-src --rosdistro ${ROS_DISTRO} -y
+
+# Set up the entrypoint
 COPY .docker/ros_entrypoint.sh /sbin/ros_entrypoint.sh
-RUN sudo chmod 755 /sbin/ros_entrypoint.sh
+RUN echo "source /sbin/ros_entrypoint.sh" >> ~/.bashrc
```

Comment:
This makes it so that even the extra shells created with `docker exec -it ...` get the workspace sourced.